book/src/tooling/coap.md

```diff
@@ -0,0 +1,242 @@
+and a path `/fwup` for interacting with firmware updates.
+The CoAP implementation can put additional resources at well-known locations,
+eg. `/.well-known/core` for discovery or `/.well-known/edhoc` for establishing secure connections.
+
+The handler does not need to concern itself with authorization:
+The CoAP stack is configured with an access policy,
+which is evaluated before the handler even receives the request.
+The handler still needs to concern itself with security to some extent
+(eg. file format parsers should still treat incoming data as possibly malformed),
+but is not the main enforcement point.
+
+Policy is configured depending on the enabled security mechanisms.
+Examples of described policy entries are:
+
+* This is an fixed public key, and requests authenticated with that key are allowed to GET and PUT to `/limit`.
+* The device has a shared secret from its authorization server, and any requests coming along a token from that server may do whatever the token says they may do (eg. "GET `/limit` and PUT `/led/0`").
+* Any (even unauthenticated) device GET `/hello/`.
+
+[provided as "examples/coap"]: https://github.com/future-proof-iot/RIOT-rs/tree/main/examples/coap
+[its `run()` function]: https://github.com/future-proof-iot/RIOT-rs/blob/2b76e560394884d3c8f7eaae51beefd59a316d7b/examples/coap/src/main.rs#L70
+
+### Interacting with RIOT-rs CoAP server from the host
+
+A convenient policy (which is the default of RIOT-rs's examples)
+is to allow all access to the programmer.
+When that policy is enabled in the build system,
+an unencrypted key is created the developer's [state home directory]<!-- precise location TBD -->,
+from where it can be picked up by tools such as [aiocoap-client].
+
+Furthermore,
+when a CoAP server is provisioned through the RIOT-rs build system,
+public keys and their device associations are stored
+in the developer's state home directory.
+
+Together, these files act in a similar way as the classic UNIX files `~/.netrc`, `~/.ssh/id_rsa{,.pub}`.
+They can also double as templates for an application akin to `ssh-copy-id`
+in that they enable a server policy like
+"any device previously flashed on this machine may GET all resources".
+
+[aiocoap-client]: https://aiocoap.readthedocs.io/en/latest/tools.html
+[state home directory]: https://specifications.freedesktop.org/basedir-spec/latest/
+
+## Usage: Client side
+
+The example [provided as "examples/coap"] also contains client steps.
+
+A program that triggers a CoAP request provides some components to the CoAP stack before phrasing the actual request:
+
+* A **URI describing the resource**, eg. `coap://coap.summit.riot-os.org/agenda` or `coap+tcp://[2001:db8::1]/.well-known/core`.
+
+  Note that while the address is printed here as a text URI
+  (and may even be entered as such in code),
+  it is kept in memory and sent as binary components.
+
+* Optionally, directions how to reach or find the host.
```

Comment:
From @kaspar030: Is this fact or opinion?

Note to self: elaborate.

Reply:
While URI is kind of self-evident to readers making the connection from HTTP, the others are not.

I *claim* that 2 and 3 are factual requirements, which the relevant standards just failed to spell out. Libraries may make it non-obvious because they make the choice for you -- for example, a browser has a "proxy" configuration field and a "do you want to use DoH / which one" field that give you little but existing choice on the directions, and hardcode a "we accept X.509 certificates as long as the root CA is in the CA/BF list or the system certificates or manually added, and the name on that matches your origin". That's by far not the only model, and for example EMail does something different with DANE (nobody checks the certificate chain there, but there might be a TLSA record in DNSSec that tells which key is authorized to receive mails for that place).

It'd probably be sensible at some point to write this up, and I'm linking in some discussion in CoRE's corr-clar document (technically Carsten's document, but adoption call is underway). However, that'll take time, and if the direction makes sense to you, I'll add a footnote around "provides some components" that explains the situation.

Reply:
There is now a footnote saying that in 024f99324bde9ac42810d64ea93dca07ec0da18f